gtpv2/message/change-notification-req.go

```diff
@@ -0,0 +1,302 @@
+			return err
+		}
+		offset += ie.MarshalLen()
+	}
+	if ie := c.MEI; ie != nil {
+		if err := ie.MarshalTo(c.Payload[offset:]); err != nil {
+			return err
+		}
+		offset += ie.MarshalLen()
+	}
+	if ie := c.IndicationFlags; ie != nil {
+		if err := ie.MarshalTo(c.Payload[offset:]); err != nil {
+			return err
+		}
+		offset += ie.MarshalLen()
+	}
+	if ie := c.RATType; ie != nil {
+		if err := ie.MarshalTo(c.Payload[offset:]); err != nil {
+			return err
+		}
+		offset += ie.MarshalLen()
+	}
+	if ie := c.ULI; ie != nil {
+		if err := ie.MarshalTo(c.Payload[offset:]); err != nil {
+			return err
+		}
+		offset += ie.MarshalLen()
+	}
+	if ie := c.UCI; ie != nil {
+		if err := ie.MarshalTo(c.Payload[offset:]); err != nil {
+			return err
+		}
+		offset += ie.MarshalLen()
+	}
+	if ie := c.PGWS5S8IPAddressForControlPlane; ie != nil {
+		if err := ie.MarshalTo(c.Payload[offset:]); err != nil {
+			return err
+		}
+		offset += ie.MarshalLen()
+	}
+	if ie := c.LinkedEBI; ie != nil {
+		if err := ie.MarshalTo(c.Payload[offset:]); err != nil {
+			return err
+		}
+		offset += ie.MarshalLen()
+	}
+	for _, ie := range c.PresenceReportingAreaInformation {
+		if err := ie.MarshalTo(c.Payload[offset:]); err != nil {
+			return err
+		}
+		offset += ie.MarshalLen()
+	}
+	if ie := c.MOExceptionDataCounter; ie != nil {
+		if err := ie.MarshalTo(c.Payload[offset:]); err != nil {
+			return err
+		}
+		offset += ie.MarshalLen()
+	}
+	for _, ie := range c.SecondaryRATUsageDataReport {
+		if err := ie.MarshalTo(c.Payload[offset:]); err != nil {
+			return err
+		}
+		offset += ie.MarshalLen()
+	}
+	if ie := c.PrivateExtension; ie != nil {
+		if err := ie.MarshalTo(c.Payload[offset:]); err != nil {
+			return err
+		}
+		offset += ie.MarshalLen()
+	}
+	for _, ie := range c.AdditionalIEs {
+		if ie == nil {
+			continue
+		}
+		if err := ie.MarshalTo(c.Header.Payload[offset:]); err != nil {
+			return err
+		}
+		offset += ie.MarshalLen()
+	}
+
+	c.Header.SetLength()
+	return c.Header.MarshalTo(b)
+}
+
+// ParseChangeNotificationRequest decodes given bytes as ChangeNotificationRequest.
+func ParseChangeNotificationRequest(b []byte) (*ChangeNotificationRequest, error) {
+	c := &ChangeNotificationRequest{}
+	if err := c.UnmarshalBinary(b); err != nil {
+		return nil, err
+	}
+	return c, nil
+}
+
+// UnmarshalBinary decodes given bytes as ChangeNotificationRequest.
+func (c *ChangeNotificationRequest) UnmarshalBinary(b []byte) error {
+	var err error
+	c.Header, err = ParseHeader(b)
+	if err != nil {
+		return err
+	}
+	if len(c.Header.Payload) < 2 {
+		return nil
+	}
+
+	decodedIEs, err := ie.ParseMultiIEs(c.Header.Payload)
+	if err != nil {
+		return err
+	}
+
+	for _, i := range decodedIEs {
+		if i == nil {
+			continue
+		}
+		switch i.Type {
+		case ie.IMSI:
+			c.IMSI = i
+		case ie.MobileEquipmentIdentity:
+			c.MEI = i
+		case ie.Indication:
+			c.IndicationFlags = i
+		case ie.RATType:
+			c.RATType = i
+		case ie.UserLocationInformation:
```

Comment:
also check if `ULIForSGW` is needed here like in Create Req message, of if it can be ignored

Reply:
`ULIForSGW` not needed in this message